parallel-install/pkg/jobmanager/README.md

```diff
@@ -0,0 +1,75 @@
+- CancelContext are being propagated to give developers the opportunity to cancel deploy.
+
+
+- This mechanism supports jobs for two different use cases: The __component-based__ jobs and the __global/component-independent__ jobs
+  - __Component-based__:
+    - Check whether the component is installed on the cluster or must be newly installed; and only trigger if it must be installed.
+    - It is possible to trigger jobs before and after a deployment of a component.
+  - __Global / Component-independent__:
+    - Always trigger the job when installing or upgrading Kyma.
+    - It is to trigger jobs before and after the deployment of Kyma.
+    - Call component-independent jobs `global` jobs to stick to the naming convention of our helm charts.
+
+### How does it work?
+
+This package registers, manages, and triggers certain jobs to have a fully-automated installation or migration. This package has two (hash)maps to manage the workload: One for `pre`-jobs and one for `post`-jobs. In the (hash)maps, the key is the name of the component the jobs belong to, and the value is a slice of the jobs.
+
+Furthermore, the `jobManager` package has a `duration` variable for benchmarking, which will be fetched at the end of a Kyma deployment.
+
+Jobs are implemented within the `jobManager` package in `go`-files, one for each component, using the specific `job` interface. Then, the implemented interface is registered using `register(job)` in the same file.
+To implement the `job` interface, the newly created jobs must implement the `execute(*config.Config, kubernetes.Interface)` function, which takes the installation config and a kubernetes interface as input, so that the jobs can interact with the cluster. The return value must be an error. Additionally, the `when()` function must be implemented, which returns the component the job is bound to and whether it should run pre or post the deployment. The `identify()` function also needs to be implemented to have a unique identifier for each job. If the active solution for tagging jobs as deprecated is chosen, then the `deprecate` function also must be implemented - more in the next section.
+
+The jobManager is used by the `deployment` package and in the `engine` package . At the hooks, during the deployment phase, each hook only has to check if the key for the wanted component is present in the pre- or post-map. If it's present, the jobs in the map are trigged, if not, nothing must be done.
+
+Retries for the jobs are not handled by the jobManager. Retries should be implemented by the jobs themselves, because it's more flexible and the interface is easy to manage. Also, the check if the logic of the job should be executed stays inside of the job, and is not implemented by the jobManager.
+
+### Deprecation of Jobs
+
+At this point, we will stick to the passiv solution, we will may switch to the active solution in the future. If no deprecation version is decided or not necessary, should also be documented.
```

Comment:
"will" or "may" ?

```suggestion
At this point, we stick to the passiv solution; we may switch to the active solution in the future. The deprecation version must be documented; alternatively, the fact that there is no specific deprecation version or that deprecation is unnecessary.
```
But doesn't the 2nd sentence actually belong to the bullet list of requirements below the following Passive/Acvice headlines?

Reply:
I can also add it there if you want to